src/App.vue

```diff
@@ -736,98 +711,113 @@ export default class App extends Vue {
     }
   }
 
-  /** Fetches and stores the business data. */
-  async fetchBusinessData (): Promise<void> {
-    const data = await LegalServices.fetchBusinessInfo(this.getBusinessId)
-
-    this.storeBusinessInfo(data)
-  }
-
-  /** Fetches draft dissolution and returns the resources. */
-  private async handleDraftDissolution (): Promise<DissolutionResourceIF> {
+  /** Fetches draft Dissolution or Restoration and sets the resources. */
+  private async handleDissolutionOrRestoration (businessId: string): Promise<void> {
```

Comment:
Everything in here looks very similar to handleIaOrRegistration() below, except:
- draft is fetched as a task instead of filing
- handles dissolution/restoration instead of IA/registration
- doesn't have NR validation (but I think it will need it)